common/test/acceptance/tests/studio/test_studio_general.py

```diff
@@ -62,6 +71,145 @@ def test_dashboard_no_courses(self):
+        )
+        self.user = None
+
+    def install_course_fixture(self):
+        """
+        Install a course fixture
+        """
+        self.course_fixture.install()
+        self.user = self.course_fixture.user
+
+    def test_sign_up_from_home(self):
+        """
+        Scenario: Sign up from the homepage
+        Given I visit the Studio homepage
+        When I click the link with the text "Sign Up"
+        And I fill in the registration form
+        And I press the Create My Account button on the registration form
+        Then I should see an email verification prompt
+        """
+        # HowitworksPage is the home page when user
+        # is not logged in. Visit it.
+        home_page = HowitworksPage(self.browser)
+        home_page.visit()
+        # Click sign up
+        click_css(page=home_page, css='.action.action-signup', source_index=0, require_notification=False)
+        unique_number = uuid.uuid4().hex[:4]
+        registration_dic = {
+            '#email': '{}-email@host.com'.format(unique_number),
+            '#name': '{}-name'.format(unique_number),
+            '#username': '{}-username'.format(unique_number),
+            '#password': '{}-password'.format(unique_number),
+        }
+        self.sign_up_page.wait_for_page()
+        # Register the user.
+        self.sign_up_page.sign_up_user(registration_dic)
+        home = HomePage(self.browser)
+        home.wait_for_page()
+
+    def test_login_with_valid_redirect(self):
+        """
+        Scenario: Login with a valid redirect
+        Given I have opened a new course in Studio
+        And I am not logged in
+        And I visit the url "/course/slashes:MITx+999+Robot_Super_Course"
+        And I should see that the path is "/signin?next=/course/slashes%3AMITx%2B999%2BRobot_Super_Course"
+        When I fill in and submit the signin form
+        Then I should see that the path is "/course/slashes:MITx+999+Robot_Super_Course"
+        """
+        self.install_course_fixture()
+        # Get the url, browser should land here after sign in.
+        course_url = self.course_outline_sign_in_redirect_page.url
+        self.course_outline_sign_in_redirect_page.visit()
+        # Login
+        self.login_page.login(self.user['email'], self.user['password'])
+        self.course_outline_page.wait_for_page()
+        # Verify that correct course is displayed after sign in.
+        self.assertEqual(self.browser.current_url, course_url)
+
+    def test_login_with_invalid_redirect(self):
+        """
+        Scenario: Login with an invalid redirect
+        Given I have opened a new course in Studio
+        And I am not logged in
+        And I visit the url "/signin?next=http://www.google.com/"
+        When I fill in and submit the signin form
+        Then I should see that the path is "/home/"
+        """
+        self.install_course_fixture()
+        # Visit course
+        self.course_outline_sign_in_redirect_page.visit()
```

Comment:
You need to change the redirect URL **before** visiting the page.
Something like this (not checked for syntax):
```
login_page = LoginPage
invalid_url = login_page.url + '?next=http://www.google.com'
self.browser.get(invalid_url)
```
Then assert that you are on the login page,
then fill in and submit the credentials,
then assert that you're on the home page via wait_for_page,
then assert that the path is "/home/" (not just that you are on the home page via the wait_for_page. That doesn't confirm the URL).